monitor/monitor_test.go

```diff
@@ -0,0 +1,177 @@
+package monitor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/ct-woodpecker/test"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	logKey = "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAElgyN7ptarCAX5krBwDwjhHM+b0xJjCKke+Dfr3GWSbLm3eO7muXRo8FDDdpdiRpnG4NJT0bdzq5YEer4C2eZ+g=="
+)
+
+func TestNew(t *testing.T) {
+	l := log.New(os.Stdout, "", log.LstdFlags)
+	clk := clock.NewFake()
+	clk.Set(time.Now())
+
+	logURI := "test"
+	processedLogKey := fmt.Sprintf("-----BEGIN PUBLIC KEY-----\n%s\n-----END PUBLIC KEY-----", logKey)
+	fetchDuration := time.Second
+
+	// Creating a monitor with an illegal key should fail
+	_, err := New(logURI, "⚷", fetchDuration, l, clk)
+	if err == nil {
+		t.Errorf("Expected New() with invalid key to error")
+	}
+
+	// Creating a monitor with vaild configuration should not fail
+	m, err := New(logURI, logKey, fetchDuration, l, clk)
+	if err != nil {
+		// NOTE: Using fatalf to avoid later nil pointer panics
+		t.Fatalf("Expected no error calling New(), got %s", err.Error())
+	}
+	if m == nil {
+		// NOTE: Using fatalf to avoid later nil pointer panics
+		t.Fatalf("Expected a non-nil monitor from New() when err == nil")
+	}
+
+	if m.logger != l {
+		t.Errorf("Expected monitor logger to be set to %p, got %p", l, m.logger)
+	}
+
+	if m.logURI != logURI {
+		t.Errorf("Expected monitor logURI to be %q, got %q", logURI, m.logURI)
+	}
+
+	if m.logKey != processedLogKey {
+		t.Errorf("Expected monitor logKey %q, got %q", processedLogKey, m.logKey)
+	}
+
+	if m.sthFetchInterval != fetchDuration {
+		t.Errorf("Expected monitor sthFetchDuration %s got %s", m.sthFetchInterval, fetchDuration)
+	}
+
+	if m.stats == nil {
+		t.Error("Expected monitor stats to be non-nil")
+	}
+
+	if m.client == nil {
+		t.Errorf("Expected monitor client to be non-nil")
+	}
+}
+
+// flakyClient is a type implementing the monitorCTClient interface with
```

Comment:
Let's call this `errorClient` to distinguish it from a client that *sometimes* returns an error.

Reply:
Good point. Renamed in 942b109